utils/pattern/src/frontend/mod.rs

```diff
@@ -9,17 +9,43 @@ mod serde;
 use crate::common::*;
 use crate::Error;
 use crate::PatternOrUtf8Error;
+use crate::SinglePlaceholderPattern;
 #[cfg(feature = "alloc")]
 use crate::{Parser, ParserOptions};
 #[cfg(feature = "alloc")]
 use alloc::{borrow::ToOwned, str::FromStr, string::String};
+use core::ops::Deref;
 use core::{
     convert::Infallible,
     fmt::{self, Write},
     marker::PhantomData,
 };
 use writeable::{adapters::TryWriteableInfallibleAsWriteable, PartsWrite, TryWriteable, Writeable};
 
+#[allow(clippy::unwrap_used, clippy::indexing_slicing)]
+impl<Store> SinglePlaceholderPattern<Store> where Store: Deref<Target = str> {
```

Comment:
@sffc how do I do this nicely? Are there encoding helpers?

Reply:
Use `Pattern::try_from_items` and then `Pattern::take_store`